pkg/controllers/customroute/custom_route_controller.go

```diff
@@ -225,31 +223,45 @@ func (c *customRouteController) updateIngressConfigStatus(ctx context.Context, i
 		return err
 	}
 
-	componentRoute := &configv1.ComponentRouteStatus{
-		Namespace:        c.componentRoute.Namespace,
-		Name:             c.componentRoute.Name,
-		DefaultHostname:  configv1.Hostname("oauth-openshift." + ingressConfig.Spec.Domain),
-		CurrentHostnames: []configv1.Hostname{configv1.Hostname(route.Spec.Host)},
-		ConsumingUsers: []configv1.ConsumingUser{
-			"system:serviceaccount:oauth-openshift:authentication-operator",
-		},
-		RelatedObjects: []configv1.ObjectReference{
-			{
-				Group:     routev1.GroupName,
-				Resource:  "routes",
-				Name:      "oauth-openshift",
-				Namespace: "openshift-authentication"},
-		},
+	componentRoute := applyconfigv1.ComponentRouteStatus().
+		WithNamespace(c.componentRoute.Namespace).
+		WithName(c.componentRoute.Name).
+		WithDefaultHostname(configv1.Hostname("oauth-openshift." + ingressConfig.Spec.Domain)).
+		WithCurrentHostnames(configv1.Hostname(route.Spec.Host)).
+		WithConsumingUsers("system:serviceaccount:oauth-openshift:authentication-operator").
+		WithRelatedObjects(
+			applyconfigv1.ObjectReference().
+				WithNamespace("openshift-authentication").
+				WithName("oauth-openshift").
+				WithGroup(routev1.GroupName).
+				WithResource("routes"),
+		)
+
+	newConditions := checkErrorsConfiguringCustomRoute(customRouteErrors)
+	if newConditions == nil {
+		newConditions = checkIngressURI(ingressConfig, route)
+		if newConditions == nil {
+			newConditions = checkRouteAvailablity(c.secretLister, ingressConfig, route)
+		}
 	}
-	conditions := checkErrorsConfiguringCustomRoute(customRouteErrors)
-	if conditions == nil {
-		conditions = checkIngressURI(ingressConfig, route)
-		if conditions == nil {
-			conditions = checkRouteAvailablity(c.secretLister, ingressConfig, route)
+	newConditions = ensureDefaultConditions(newConditions)
+	// set timestamps to last transitioned if available, otherwise, write a new lasttransitioned.
+	// this may not handle extremely rapid flapping well.
+	if existingComponentRoute := common.GetComponentRouteStatus(ingressConfig, c.componentRoute.Namespace, c.componentRoute.Name); existingComponentRoute != nil {
+		for i := range newConditions {
+			newCondition := newConditions[i]
+			if existingCondition := v1helpers.FindCondition(existingComponentRoute.Conditions, newCondition.Type); existingCondition != nil {
+				if newCondition.Status == existingCondition.Status {
+					newConditions[i].LastTransitionTime = existingCondition.LastTransitionTime
+				}
+			}
 		}
 	}
-	componentRoute.Conditions = ensureDefaultConditions(conditions)
-	_, err = c.updateComponentRouteStatus(ctx, componentRoute)
+
+	componentRoute.WithConditions(newConditions...)
+
+	ingressStatus := applyconfigv1.Ingress(ingressConfig.Name).WithStatus(applyconfigv1.IngressStatus().WithComponentRoutes(componentRoute))
+	_, err = c.ingressClient.ApplyStatus(ctx, ingressStatus, c.forceApply())
```

Comment:
@deads2k  is it safe to assume that SSA is always enabled on the server?

Reply:
> @deads2k is it safe to assume that SSA is always enabled on the server?

Yes